plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -61,6 +65,49 @@
 
     public static final String TYPE = "s3";
 
+    // prefix for aws client settings
+    private static final String PREFIX = "aws.config.";
```

Comment:
> That has been our policy since the beginning

I remember that long ago, prior to 2.0. There used to be crazy support for this in setting names, even reflecting on the package to come up with setting names IIRC. But that was all removed a long time ago.